src/psyclone/transformations.py

```diff
@@ -3836,24 +3866,23 @@ def apply(self, node, options=None):
         :param options: a dictionary with options for transformations.
         :type options: dictionary of string:values or None
 
-        :returns: tuple of the modified Schedule and a record of the \
-                  transformation.
-        :rtype: (:py:class:`psyclone.psyGen.Schedule`, \
-                 :py:class:`psyclone.undoredo.Memento`).
+        :returns: a record of the transformation.
```

Comment:
Returns the schedule too.

Reply:
Not any more (#595 )

Reply:
That issue raises a question that hasn't been agreed!

Reply:
@hiker and I agree :-)

Reply:
Well I think I agree too and we could also throw away memento given that it is not used and that the same functionality could be supported externally to transformations (but less efficiently) by copying the full PSyIR tree before each transformation. However, Sergi and Iva should also be asked for their input before making a decision, the question should be changed into a standard issue and the issue should be referenced here so people know why this transformation is different to the rest.

Reply:
OK, I've asked Sergi and Iva so we'll have to wait and see. Alternatively, being the impatient person that I am I could put it back to the way it used to be and then we change all transformations once we have agreed.

Reply:
A separation of concerns - it could catch on :-)